test/tests/functional/pbs_multi_sched.py

```diff
@@ -579,3 +579,93 @@ def test_run_limts_per_scheduler(self):
+                                                     default_shares)
+        self.scheds['default'].set_fairshare_usage(TEST_USER, default_usage)
+
+        self.scheds['sc1'].add_to_resource_group(TEST_USER, 10, 'root',
+                                                 sc1_shares)
+        self.scheds['sc1'].set_fairshare_usage(TEST_USER, sc1_usage)
+
+        self.scheds['sc2'].add_to_resource_group(TEST_USER, 10, 'root',
+                                                 sc2_shares)
+        self.scheds['sc2'].set_fairshare_usage(TEST_USER, sc2_usage)
+
+        self.scheds['sc3'].add_to_resource_group(TEST_USER, 10, 'root',
+                                                 sc3_shares)
+        self.scheds['sc3'].set_fairshare_usage(TEST_USER, sc3_usage)
+
+        # requery fairshare info from pbsfs
+        default_fs = self.scheds['default'].query_fairshare()
+        sc1_fs = self.scheds['sc1'].query_fairshare()
+        sc2_fs = self.scheds['sc2'].query_fairshare()
+        sc3_fs = self.scheds['sc3'].query_fairshare()
+
+        n = default_fs.get_node(id=10)
+        self.assertEquals(n.nshares, default_shares)
+        self.assertEquals(n.usage, default_usage)
+
+        n = sc1_fs.get_node(id=10)
+        self.assertEquals(n.nshares, sc1_shares)
+        self.assertEquals(n.usage, sc1_usage)
+
+        n = sc2_fs.get_node(id=10)
+        self.assertEquals(n.nshares, sc2_shares)
+        self.assertEquals(n.usage, sc2_usage)
+
+        n = sc3_fs.get_node(id=10)
+        self.assertEquals(n.nshares, sc3_shares)
+        self.assertEquals(n.usage, sc3_usage)
+
+    def test_sched_priv_change(self):
+        """
+        Test that when the sched_priv directory changes, all of the 
+        PTL internal scheduler objects (e.g. fairshare tree) are reread
+        """
+
+        new_sched_priv = os.path.join(self.server.pbs_conf['PBS_HOME'],
+                                      'sched_priv2')
+        if os.path.exists(new_sched_priv):
+            self.du.rm(path=new_sched_priv, recursive=True,
+                       sudo=True, force=True)
+
+        dflt_sched_priv = os.path.join(self.server.pbs_conf['PBS_HOME'],
+                                       'sched_priv')
+
+        self.du.run_copy(src=dflt_sched_priv, dest=new_sched_priv,
+                         recursive=True, sudo=True)
+        self.setup_sc3()
+        self.scheds['sc3'].add_to_resource_group(TEST_USER, 10, 'root', 20)
+        self.scheds['sc3'].holidays_set_year(new_year="3000")
+
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'sched_priv': new_sched_priv}, id='sc3')
+
+        n = self.scheds['sc3'].fairshare_tree.get_node(id=10)
+        print n
+        self.assertFalse(n)
+
+        y = self.scheds['sc3'].holidays_get_year()
+        self.assertNotEquals(y, "3000")
+
+        # clean up
+        self.du.rm(path=new_sched_priv, sudo=True, recursive=True, force=True)
```

Comment:
revert_to_defaults will do the cleanup, at the end of the test. Why is this required?